apps/cart/views.py

```diff
@@ -0,0 +1,57 @@
+from django.shortcuts import render
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.decorators import action
+from rest_framework import status
+from rest_framework.response import Response
+
+from apps.cart.models import CartItem, ShoppingCart
+from apps.cart.serializers import CartItemSerializer, ShoppingCartSerializer
+from apps.products.models import Product
+
+
+class ShoppingCartViewSet(ModelViewSet):
+    serializer_class = ShoppingCartSerializer
+
+    http_method_names = [m for m in ModelViewSet.http_method_names if m not in ["put", "patch"]]
+
+    def get_queryset(self):
+        return ShoppingCart.objects.filter(user=self.request.user)
+
+    def get_object(self):
+        obj, created = ShoppingCart.objects.get_or_create(user=self.request.user)
+        return obj
+
+    @action(detail=False, methods=['post'], url_path='add/(?P<product_id>[^/.]+)')
+    def add_to_cart(self, request, product_id=None):
+        try:
+            product = Product.objects.get(id=product_id)
+
```

Comment:
This is a post request so why dont you send the data as a post data.... it will help in sending more data like the quantity too on create.... instead of making multiple request to increase the quantity

Reply:
You want me to allow the user to put in the quantity they want?

Reply:
yes but set the default as 1